wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -0,0 +1,513 @@
+      .build()
+  }
+
+  // TODO add support for custom adapters.
+  private fun getAdapterName(field: Field): CodeBlock {
+    if (field.type().isScalar) {
+      return CodeBlock.of("%T.%L", ProtoAdapter::class,
+        field.type().simpleName().toUpperCase(Locale.US))
+    }
+
+    return CodeBlock.of("%L.ADAPTER", typeName(field.type()).simpleName)
+  }
+
+  /**
+   * ```
+   * Example
+   * enum class PhoneType(private val value: Int) : WireEnum {
+   *     HOME(0),
+   *     ...
+   *     override fun getValue(): Int = value
+   *     object ADAPTER { ... }
+   * ```
+   * }
+   */
+  private fun generateEnum(message: EnumType): TypeSpec {
+    val type = message.type()
+    val nameAllocator = nameAllocator(message)
+
+    val valueName = nameAllocator.get("value")
+
+    var builder = TypeSpec.enumBuilder(type.simpleName())
+      .addSuperinterface(WireEnum::class)
+      .primaryConstructor(FunSpec.constructorBuilder()
+        .addParameter(valueName, Int::class, PRIVATE)
+        .build())
+      .addProperty(PropertySpec.builder(valueName, Int::class, PRIVATE)
+        .initializer(valueName)
+        .build())
+      .addFunction(FunSpec.builder("getValue")
+        .returns(Int::class)
+        .addModifiers(OVERRIDE)
+        .addStatement("return $valueName")
+        .build())
+      .addType(generateEnumAdapter(message))
+
+    message.constants().forEach { constant ->
+      builder.addEnumConstant(nameAllocator.get(constant), TypeSpec.anonymousClassBuilder()
+        .addSuperclassConstructorParameter("%L", constant.tag())
+        .addKdoc(constant.documentation())
+        .build())
+    }
+
+    return builder.build()
+  }
+
+  /**
+   * Example
+   * ```
+   * object ADAPTER : EnumAdapter<PhoneType>(PhoneType::class.java) {
+   *     override fun fromValue(value: Int): PhoneType? = values().find { it.value == value }
+   * }
+   * ```
+   */
+
+  private fun generateEnumAdapter(message: EnumType): TypeSpec {
+    val parentClassName = nameToKotlinName.getValue(message.type())
+    val nameAllocator = nameAllocator(message)
+
+    val adapterName = nameAllocator.get("ADAPTER")
+    val valueName = nameAllocator.get("value")
+
+    return TypeSpec.objectBuilder(adapterName)
+      .superclass(EnumAdapter::class.asClassName().parameterizedBy(parentClassName))
+      .addSuperclassConstructorParameter("%T::class.java", parentClassName)
+      .addFunction(FunSpec.builder("fromValue")
+        .addModifiers(OVERRIDE)
+        .addParameter(valueName, Int::class)
+        .returns(parentClassName.asNullable())
+        .addStatement("return values().find { it.value == value }")
+        .build())
+      .build()
+  }
+
+  /**
+   * Adds code to help message class implement Parcelable.
+   */
+  private fun addAndroidCodeToMessage(classBuilder: TypeSpec.Builder, type: MessageType) {
+    classBuilder.addFunction(FunSpec.builder("writeToParcel")
+      .addStatement("return destination.writeByteArray(ADAPTER.encode(this))")
```

Comment:
I believe the formatting is off in this file.. We need double-indents for chained calls

Reply:
This is how it looks like this right now https://github.com/square/wire/pull/700/files#diff-229360ca3bee88f73e53e4ec2eeab474R30